pkg/chain/ethereum/tbtc.go

```diff
@@ -103,11 +108,38 @@ func newTbtcChain(
 		)
 	}
 
+	lightRelayAddress, err := config.ContractAddress(LightRelayContractName)
```

Comment:
This `LightRelay` contract address extraction requires that we have `LightRelayAddress` defined in our configs when starting `keep-client`.

Reply:
I do not think this is an acceptable solution. It is possible to start the client on mainnet without a config file, [just by providing the required parameters in CLI](https://docs.threshold.network/staking-and-running-a-node/running-a-node/self-managed/tbtc-v2-node-setup/binary-installation#configure-tbtc-v2-client) and even if there is a config file, `LightRelay` is not there.

Since we are not pulling `tbtc-v2` contract addresses from mainnet NPM package (see #3408), all clients running on mainnet would fail to start after the upgrade, no matter if they use config file or just CLI parameters.

One option would be to move the difficulty-related functions to a separate chain structure but I do not think it is worth the complications, especially that this functionality is tightly-coupled with tBTC. 

A quick fix could be to:
1. Do not fail here if the address is not configured, just don't create the light relay contract.
2. For every light relay function, check if the contract was instantiated. If not, report an error.

In the code reading the config, we'd need to check whether the bitcoin difficulty maintainer is enabled and if it is enabled, then whether the ethereum address for the `LightRelay` contract is configured.

Reply:
I think we could validate it [here](https://github.com/keep-network/keep-core/blob/fc2b0d36d5b09fd770c0c4769acc31839aa02ce1/config/config.go#L154-L187). Currently, we do not do a cross-section validations, but I think doing so is perfectly acceptable.

Reply:
One thing to note is that we could set the `LightRelay` address via command line option:
`--developer.lightRelayAddress 0x0000000000000000000000000000000000000000`

Reply:
It is true and I think this would be acceptable for development environments but this is a breaking change for production configs and something that would be temporary - we will resolve the addresses from the package at some point.

Reply:
I moved the Bitcoin difficulty-related functionalities to a separate chain in 631768e43339d9a2481810fbcfb4614326d142c2.

Therefore there is no longer need to set any value for `LightRelayAddress` when starting the keep-client for keep nodes running.

Reply:
Nice, that makes sense!